torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -503,7 +503,7 @@ class TestDistBackend(MultiProcessTestCase):
     @classmethod
     def setUpClass(cls):
         os.environ["MASTER_ADDR"] = str(MASTER_ADDR)
-        os.environ["MASTER_PORT"] = str(MASTER_PORT)
+        # Not setting MASTER_PORT and get a random free port
```

Comment:
@wanchaol @huydhn wonder what is the process to find the free port, is there a `find_free_port` call somewhere? If so, how do we ensure safety across processes now that we're running multiple backends in parallel?

Reply:
AFAIK, not setting a port here means getting a free port because that's the behavior of binding to port 0 https://stackoverflow.com/questions/1075399/how-to-bind-to-any-available-port. This behavior is used throughout the code `s.bind(("localhost", 0))` when no port is set and that's also how find_free_port works.  Looking closer into the details, I learn that there is a potential race condition here in which the free port might be picked up before the process can use it:

* https://github.com/pytorch/pytorch/blob/master/torch/distributed/elastic/agent/server/api.py#L363-L364
* https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_utils.py#L2825-L2828

Do you have any thoughts on how `retry_on_connect_failures` is used throughout distributed test suite?